### PR TITLE
Added biome fog

### PR DESCRIPTION
**Pull request by OneAvargeCoder193**

### CODE REVIEW

File: src/game.zig [hunk 1]
```diff
@@ -325,5 +326,14 @@ pub fn update(deltaTime: f64) void {
 		defer Player.mutex.unlock();
 		Player.super.pos += movement*@as(Vec3d, @splat(deltaTime));
 	}
+
+	const biome = world.?.playerBiome.load(.seq_cst);
```
Comment (IntegratedQuantum) [inline src/game.zig:330]: .monotonic is enough. Sequential consistency just makes it more expensive for no reason

File: src/game.zig [hunk 2]
```diff
@@ -325,5 +326,14 @@ pub fn update(deltaTime: f64) void {
...
+	// std.log.debug("r {d}, g {d}, b {d}", .{biome.fogColor[0], biome.fogColor[1], biome.fogColor[2]});
```
Comment (IntegratedQuantum) [inline src/game.zig:331]: Please remove this debug statement.

File: assets/cubyz/biomes/desert.json [hunk 3]
```diff
@@ -9,6 +9,9 @@
 	"hills" : 20,
 	
 	"music" : "EasternThought",
+
+	"fogDensity" : 20,
+	"fogColor" : 0xffe0dcc3,
```
Comment (IntegratedQuantum) [inline assets/cubyz/biomes/desert.json:14]: Why the extra ff in front of it? The alpha component isn't even used, right?

File: src/renderer.zig [hunk 4]
```diff
@@ -132,7 +132,7 @@ pub fn render(playerPosition: Vec3d) void {
 		const skyColor = vec.xyz(world.clearColor);
 		game.fog.color = skyColor;
 
-		renderWorld(world, ambient, skyColor, playerPosition);
+		renderWorld(world, ambient, game.fog.color, playerPosition);
```
Comment (IntegratedQuantum) [inline src/renderer.zig:135]: Why this change? Those are the same values anyways.

File: src/graphics.zig [hunk 5]
```diff
@@ -1912,6 +1912,7 @@ pub const Image = struct {
 };
 
 pub const Fog = struct {
+	fogColor: Vec3f,
 	color: Vec3f,
```
Comment (IntegratedQuantum) [inline src/graphics.zig:1916]: Could you maybe improve the names here?
fog.color and fog.fogColor convey the same meaning to me.

**Review by IntegratedQuantum — COMMENTED** [review]
Looks good, I'd like you to just make a few small changes.

File: src/game.zig [hunk 6]
```diff
@@ -327,8 +327,7 @@ pub fn update(deltaTime: f64) void {
 		Player.super.pos += movement*@as(Vec3d, @splat(deltaTime));
 	}
 
-	const biome = world.?.playerBiome.load(.seq_cst);
-	// std.log.debug("r {d}, g {d}, b {d}", .{biome.fogColor[0], biome.fogColor[1], biome.fogColor[2]});
+	const biome = world.?.playerBiome.monotonic;
```
Comment (IntegratedQuantum) [inline src/game.zig:330]: No, this can't compile. Should be .load(.monotonic)

**Comment by OneAvargeCoder193** [comment]
That should work

File: saves/Development/palette.json [hunk 7]
```diff
@@ -1,69 +1,120 @@
...
 	"cubyz:glass/white" : 62,
-	"cubyz:cactus" : 63,
-	"cubyz:sandstone" : 64,
+	"cubyz:chalk/indigo" : 115,
 	"cubyz:oak_leaves" : 65,
-	"cubyz:glow_crystal/dark_gray" : 66
+	"cubyz:glow_crystal/dark_gray" : 66,
+	"cubyz:chalk/black" : 116,
+	"cubyz:chalk/aqua" : 117
 }
```
Comment (IntegratedQuantum) [inline saves/Development/palette.json:120]: Well, it seems like you accidentally commited some world data.

**Comment by OneAvargeCoder193** [comment]
I fixed the world
